llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2436,13 +2438,20 @@ void VPlanTransforms::handleUncountableEarlyExit(
       ExitIRI->extractLastLaneOfOperand(MiddleBuilder);
     }
     // Add the incoming value from the early exit.
-    if (!IncomingFromEarlyExit->isLiveIn() && !Plan.hasScalarVFOnly()) {
-      VPValue *FirstActiveLane = EarlyExitB.createNaryOp(
-          VPInstruction::FirstActiveLane, {EarlyExitTakenCond}, nullptr,
-          "first.active.lane");
-      IncomingFromEarlyExit = EarlyExitB.createNaryOp(
-          Instruction::ExtractElement, {IncomingFromEarlyExit, FirstActiveLane},
-          nullptr, "early.exit.value");
+    if (!IncomingFromEarlyExit->isLiveIn()) {
+      // Limit range to scalar VF only, if the range contains the scalar VF.
+      bool IsVectorVF = LoopVectorizationPlanner::getDecisionAndClampRange(
+          [&](ElementCount VF) { return VF.isVector(); }, Range);
```

Comment:
```suggestion
          [](ElementCount VF) { return VF.isVector(); }, Range);
```

No need to capture anything I think?

Reply:
Done, thanks.